test/src/com/xilinx/rapidwright/eco/TestECOTools.java

```diff
@@ -0,0 +1,438 @@
+            }
+            sourceSitePinInsts.add(sourcePins);
+            sinkSitePinInsts.put(net, sinkPins);
+
+            Assertions.assertNotNull(ehpi);
+            Assertions.assertTrue(ehpi.isOutput());
+            disconnectPins.add(ehpi);
+        }
+        ECOTools.disconnectNet(design, disconnectPins, deferredRemovals);
+        Assertions.assertEquals(2, deferredRemovals.size());
+
+        // Swap those output pins
+        Map<EDIFHierNet, List<EDIFHierPortInst>> netToPortInsts = new HashMap<>();
+        netToPortInsts.put(disconnectedNets.get(0), new ArrayList() {{ add(disconnectPins.get(1)); }});
+        netToPortInsts.put(disconnectedNets.get(1), new ArrayList() {{ add(disconnectPins.get(0)); }});
+
+        ECOTools.connectNet(design, netToPortInsts, deferredRemovals);
+        Assertions.assertEquals(0, deferredRemovals.size());
+
+        List<Net> physNets = new ArrayList<>();
+        for (EDIFHierNet ehn : disconnectedNets) {
+            Net net = design.getNet(ehn.getHierarchicalNetName());
+            physNets.add(net);
+            Assertions.assertFalse(net.hasPIPs());
+            Set<SitePinInst> sinkPins = sinkSitePinInsts.get(net);
+            for (SitePinInst spi : net.getPins()) {
+                Assertions.assertFalse(spi.isRouted());
+                // Check sink SPIs are not swapped
+                Assertions.assertTrue(spi.isOutPin() || sinkPins.contains(spi));
+            }
+        }
+
+        // Check source SPI is swapped
+        Assertions.assertTrue(sourceSitePinInsts.get(0).contains(physNets.get(1).getSource().getSitePinName()));
+        Assertions.assertTrue(sourceSitePinInsts.get(1).contains(physNets.get(0).getSource().getSitePinName()));
+
+        if (FileTools.isVivadoOnPath()) {
+            ReportRouteStatusResult rrs = VivadoTools.reportRouteStatus(design);
+            Assertions.assertEquals(0, rrs.netsWithRoutingErrors);
+            Assertions.assertEquals(2, rrs.unroutedNets);
+        }
+    }
+
+    @Test
+    @Disabled("Currently, ECOTools.removeCell() does not work for hierarchical cells. Specifically, for this testcase " +
+            "exclusively intra-site routes (e.g. 'processor/data_path_loop[4].small_spm.small_spm_ram.spm_ram/DOA') " +
+            "are not removed and appear in the DCP causing Vivado to emit 'placement information for XX sites failed to" +
+            "restore' warnings and cells (e.g. 'your_program/ram_4096x8') to be unplaced.")
+    public void testRemoveCell() {
+        Design design = RapidWrightDCP.loadDCP("picoblaze_ooc_X10Y235.dcp");
+        EDIFNetlist netlist = design.getNetlist();
+        Map<Net, Set<SitePinInst>> deferredRemovals = new HashMap<>();
+
+        EDIFHierCellInst ehciToDelete = netlist.getHierCellInstFromName("processor");
+        List<EDIFHierCellInst> leavesToDelete = netlist.getAllLeafDescendants(ehciToDelete);
+
+        ECOTools.removeCell(design, Collections.singletonList(ehciToDelete), deferredRemovals);
```

Comment:
It seems `ECOTools.removeCell()` returns an object.  Should we capture the return value and also test it?

Reply:
`ECOTools.removeCell()` does not return anything.

Reply:
Sorry, I was relying on GitHub's context menu and realized they are referring to another class.